### PR TITLE
crypto/omni_blockchain_subscription_test

### PR DESCRIPTION
**Pull request by binary-tanya**

### CODE REVIEW

File: t/ust.t [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+            else { $result = []; }
+
+            return Future->done($result);
+        },
+        get_transaction => sub {
+            my ($self, @params) = @_;
+            my $omni_gettransaction = {
+                blocktime        => 1582166123,
+                confirmations    => 1,
+                blockhash        => '00000000000b031f58f9761fd1e0219ef5178481f274b4f54d30c5142b97571f',
```
Comment (chylli-binary) [inline t/ust.t]: can we make this one, and 'from' 'to' address as variables ? The const string is so long and error-prone.

Comment (chylli-binary) [reply t/ust.t]: And  variable will make code more clear and understandable.

File: t/ust.t [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            $subscription_source->finish();
+        });
+
+    my $emitted_transaction = $subscription_client->hashblock('0000000027072628ae58009d904e586cce12f2c68d3a501cbb0a781555bcb5d2')->get;
+
+    $mocked_omni->unmock_all();
+    $mocked_rpc_omni->unmock_all();
+};
+
+done_testing();
```
Comment (Binary-Mohanad) [inline t/ust.t]: You should add a test for the internal transactions as well

**Comment by binary-tanya** [comment]
Could we add transaction data for LTC too since we are using BTC.pm itself for both crypto currencies. ?

File: t/ust.t (hunk 1 above)
Comment (binary-tanya) [reply t/ust.t]: Updated the strings with variable . 👍

File: t/ust.t (hunk 2 above)
Comment (binary-tanya) [reply t/ust.t]: Added. 👍

File: t/ust.t [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+            my $transaction = shift;
+            is $transaction->{currency}, 'UST',  "Currency code: $transaction->{currency} is correct";
+            is $transaction->{amount},   "21",   "Amount is correct";
+            is $transaction->{type},     'sent', "Transaction Type: $transaction->{type} is correct";
+            is_deeply $transaction, $expected_transaction, "$currency_code Omni-Send Transaction is emitted correctly.";
+            $subscription_source->finish();
+        });
+
+    my $emitted_transaction = $subscription_client->hashblock($blockhash)->get;
+
```
Comment (Binary-Mohanad) [inline t/ust.t]: you should call here `$subscription_source->get;` to make sure the future has been ended

File: t/ust.t [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+            my $transaction = shift;
+            is $transaction->{currency}, 'UST',     "Currency code: $transaction->{currency} is correct";
+            is $transaction->{amount},   "56",      "Amount is correct";
+            is $transaction->{type},     'receive', "Transaction Type: $transaction->{type} is correct";
+            is_deeply $transaction, $expected_transaction, "$currency_code Omni-Send All Transaction is emitted correctly.";
+            $subscription_source->finish();
+        });
+
+    my $emitted_transaction = $subscription_client->hashblock($blockhash)->get;
+
```
Comment (Binary-Mohanad) [inline t/ust.t]: Also here  `$subscription_source->get;`

File: t/ust.t [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+    my $hash = '19e5bda81d0a588cbf46072fb6da01685a51082d4b94428e09f491ad8a111bc7';
+    my $blockhash = '00000000000b031f58f9761fd1e0219ef5178481f274b4f54d30c5142b97571f';
+    
+    my $expected_transaction = Net::Async::Blockchain::Transaction->new(
+        currency     => $currency_code,
+        hash         => $hash,
+        block        => '1666276',
+        from         => $from,
+        to           => $to,
+        amount       => Math::BigFloat->new(21),
```
Comment (michaelmueller-binary) [inline t/ust.t]: You don't seem to `use` Math::BigFloat`  before using it.

Comment (binary-tanya) [reply t/ust.t]: We are declaring it here https://github.com/binary-com/perl-Net-Async-Blockchain/blob/master/lib/Net/Async/Blockchain/Omni.pm#L42. So it's usable without declaring it...

**Pull request merged**